### ISSUE TITLE
Livelock

### ISSUE THREAD

**Issue by abhishekgupta-1** [abhishekgupta-1/FUCloud#11]
Check if livelock occurs because of prevLogIndex < recoveryPrevLogIndex?

https://github.com/abhishekgupta-1/FUCloud/blob/b1101a8204ff5ba8e4b3c13a5bb5185d283d1168/server.py#L243

Consider followerId node enters the recovery node and sends the leader a message containing index of the mismatched log and updates the recoveryPrevLogIndex. If the message is lost and doesn't receive the leader, the leader will not decrement prevLogIndex and call appendEntries with the prevLogIndex value which will be >= than currently updated recoveryPrevLogIndex. In this case followerId node will do nothing.

**Comment by vidhiJain** [comment]
Why do we have this check on using prevLogIndex? Isn't this condition self satisfied? I could not find suitable explanation.
https://github.com/abhishekgupta-1/CloudHaoma/blob/b1101a8204ff5ba8e4b3c13a5bb5185d283d1168/server.py#L224

**Comment by abhishekgupta-1** [comment]
No, the prevLogIndex(starts with log size of leader and decrements on each mismatch)  is provided by the leader, the followerId may   be outdated and hence its log length may be less than prevLogIndex
